controllers/volumereplicationgroup_controller.go

```diff
@@ -817,13 +818,29 @@ func (v *VRGInstance) processAsPrimary() (ctrl.Result, error) {
 }
 
 func (v *VRGInstance) reconcileAsPrimary() ctrl.Result {
-	result := ctrl.Result{}
-	if len(v.volSyncPVCs) != 0 {
-		result.Requeue = v.reconcileVolSyncAsPrimary()
+	var finalSyncPrepared struct {
+		volSync     bool
+		kubeObjects func() bool
 	}
 
+	vrg := v.instance
+	s3StoreAccessors := s3StoreAccessorsGet(
+		vrg.Spec.S3Profiles,
+		func(s3ProfileName string) (ObjectStorer, ramendrv1alpha1.S3StoreProfile, error) {
+			return v.reconciler.ObjStoreGetter.ObjectStore(
+				v.ctx, v.reconciler.APIReader, s3ProfileName, v.namespacedName, v.log,
+			)
+		},
+		v.log,
+	)
+	result := ctrl.Result{}
+	result.Requeue = v.reconcileVolSyncAsPrimary(&finalSyncPrepared.volSync)
 	v.reconcileVolRepsAsPrimary(&result.Requeue)
-	v.kubeObjectsProtect(&result)
+	v.kubeObjectsProtect(&result, &finalSyncPrepared.kubeObjects, s3StoreAccessors)
+	v.vrgObjectProtected = vrgObjectProtect(&result, s3StoreAccessors, *vrg, v.reconciler.eventRecorder, v.log)
+	vrg.Status.PrepareForFinalSyncComplete = vrg.Spec.PrepareForFinalSync &&
```

Comment:
@hatfieldbrian, sorry to say that you are adding so much complexity with these (too many) anonymous functions. They are needed sometimes, but too many make one's head spin. It took me too long to debug this line. This boolean cannot be turned OFF once it is already ON. The reason is that the DRPC will think it needs to prepare for another final sync. Yes, we can change the DRPC to check if the final sync has been completed instead, but I think `PrepareForFinalSyncComplete` once set to True, it should stay that way.
A simple fix in this line would be:
```suggestion
        if vrg.Spec.PrepareForFinalSync && !vrg.Status.PrepareForFinalSyncComplete {
	    vrg.Status.PrepareForFinalSyncComplete = finalSyncPrepared.volSync && finalSyncPrepared.kubeObjects()
	}                                                                      
```
I am not sure if that's enough with the changes you made. Anyway, I'll give that suggestion a try...

Reply:
I tested the above suggestion and finally had a successful relocation.

Reply:
@BenamarMK Thanks for retesting.  I merged the two final sync booleans into one `finalSyncPhase` so that status need only be reported for the current phase.  `make test` passed which seems to test DRPC relocate, but I'm not sure if it tests volSync.

Reply:
Thanks @hatfieldbrian. I'll run VolSync e2e using this PR.